dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -211,3 +233,90 @@ func validateFromChainState(runtimeInstance parachainruntime.RuntimeInstance, po
 
 	return &candidateCommitments, persistedValidationData, isValid, nil
 }
+
+// validateFromExhaustive validates a candidate parachain block with provided parameters
+func validateFromExhaustive(validationHost parachainruntime.ValidationHost,
+	persistedValidationData parachaintypes.PersistedValidationData,
+	validationCode parachaintypes.ValidationCode,
+	candidateReceipt parachaintypes.CandidateReceipt, pov parachaintypes.PoV) (
+	*ValidationResult, error) {
+
+	validationCodeHash := validationCode.Hash()
+	// basic checks
+	validationErr, internalErr := performBasicChecks(&candidateReceipt.Descriptor, persistedValidationData.MaxPovSize,
+		pov,
+		validationCodeHash)
+	if validationErr != nil {
+		validationResult := &ValidationResult{
+			InvalidResult: validationErr,
+		}
+		return validationResult, internalErr
+	}
```

Comment:
Let's check for the internal error before the validation error.

What is the internal error?
 - example: encoding error, hashing error
 - If we get an internal error, that should be checked first, and internal errors should be returned.
 - if we get an internal error, it means we are not able to find the validation result.
 - So, we should return **either** a validation result **or** an internal error
 
```suggestion

	if internalErr != nil {
		return nil, fmt.Errorf("performing basic checks: %w", internalErr)
	}

	if validationErr != nil {
		validationResult := &ValidationResult{
			InvalidResult: validationErr,
		}
		return validationResult, nil
	}
```

Reply:
That makes more sense, I've updated this.